src/main/kotlin/com/neva/gradle/fork/config/properties/PropertyDefinition.kt

```diff
@@ -0,0 +1,60 @@
+package com.neva.gradle.fork.config.properties
+
+class PropertiesDefinitions {
+  private val definitions = mutableMapOf<String, PropertyDefinition>()
+
+  fun configure(propertiesConfiguration: Map<String, PropertyDefinition.() -> Unit>) {
+    definitions += propertiesConfiguration.mapValues { PropertyDefinition(it.key).apply(it.value) }
+  }
+
+  fun getProperty(prompt: PropertyPrompt): Property {
+    val definition = definitions.getOrElse(prompt.name) {
+      PropertyDefinition(prompt.name)
+    }
+    return Property(definition, prompt)
+  }
+}
+
+class PropertyDefinition(val name: String) {
+  init {
+    if (name.isBlank()) throw PropertyException("Name of property definition cannot be blank!")
+  }
+
+  val PASSWORD = PropertyType.PASSWORD
```

Comment:
what is the purpose of these vals?

Reply:
Those values are used in DSL to simplify property type specification:
`type = CHECKBOX`
instead of 
`type = com.neva.gradle.fork.config.properties.PropertyType.CHECKBOX`

Reply:
ok, but even Gradle is not doing this that way, importing enum will be not so bad and I am opting for this approach instead